.github/workflows/docker.yml

```diff
@@ -0,0 +1,58 @@
+name: Create and publish Docker images
+
+on:
+  push: { branches: [ unstable ] }
+  pull_request: { branches: [ unstable ] }
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - name: github_ci
+        #- name: jenkins_ci # flatironinstitute/cthyb:master-ubuntu-clang not accessible
+        - name: mpich
+        - name: openmpi
+        #- name: triqs_default # flatironinstitute/triqs:unstable-ubuntu-clang not accessible
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the Container registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}_${{ matrix.name }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: ./Docker
+          file: ./Docker/${{ matrix.name }}_dockerfile
+          build-args: |
+            NCORES=2
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
Can we somehow tag the docker images created here with `:unstable`, so that we can create another action for 3.2.x, in which the 3.2.x branch of all apps is build (I can take care of that).

Reply:
Images are automatically tagged by their branch name. You can try it out using
```
docker pull ghcr.io/hmenke/solid_dmft_github_ci:unstable
```